Modules/Core/Common/include/itkThreadSafeBooleanType.h

```diff
@@ -0,0 +1,63 @@
+/*=========================================================================
+ *
+ *  Copyright NumFOCUS
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0.txt
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ *=========================================================================*/
+#ifndef itkThreadSafeBooleanType_h
+#define itkThreadSafeBooleanType_h
+
+namespace itk
+{
+
+/** The type alias `ThreadSafeBooleanType` provides an alternative to `bool`. `std::vector<bool>` is not thread safe due
+ * to the possibility of multiple bits being packed together in the same memory location.
+ * std::vector<ThreadSafeBooleanType> does not have such a space optimization. std::vector<ThreadSafeBooleanType> is
+ * semantically like `std::vector<bool>`, but unlike `std::vector<bool>`, it does "avoid data races when the contents of
+ * the contained object in different elements in the same container [...] are modified concurrently", according to the
+ * C++ Standard, section [container.requirements.dataraces], "Container data races".
+ *
+ */
+
+class ThreadSafeBooleanType
```

Comment:
The class isn't entirely thread-safe (it isn't atomic), which is OK, because it just works around the infamous `std::vector<bool>` problem. However, I then find the name of the class "ThreadSafeBooleanType" too pretentious. I would prefer to just call it `Boolean`.

Note that it may also yield a performance gain in single-threaded code, as random access to `std::vector<itk::Boolean>` would probably be faster than random access to `std::vector<bool>`

Reply:
Should we use `itk::BooleanType`?

Reply:
I like `ThreadSafeBooleanType`. It gives a good indication of what this is for. Just `itk::BooleanType` is so generic, it tells nothing.

Reply:
I just pushed it as `itk::Boolean` but we could still change it.  Let me know.